test/e2e/nightfall-sdk.test.mjs

```diff
@@ -164,34 +164,50 @@ describe('Testing the Nightfall SDK', () => {
         web3,
       );
       // approve tokens to be advanced by liquidity provider in the instant withdraw
+      let txDataToSign;
       try {
-        await approve(
+        txDataToSign = await approve(
```

Comment:
Approval should only be done in bulk once per LP until the approved limit is reached and then can be done again.
Not for each time an instant withdrawal is requested because this would cost more gas

Reply:
This is just in a test. Is it really necessary to change?

Reply:
Yes, it's only to test 1 instant withdraw in the e2e test that will be transferred by the LP.

Reply:
Test files are written with dual intent - for testing purposes as well as acting as a good reference for future/other developers. In the case of this test, that would be reference for SDK and how to build dapps for Nightfall correctly.
As a result, these have to be written to good standards just as any non test code we write.

Solving this would mean simply moving the `approve` out of event handling code to the phase when we register this LP.

Reply:
You are right. `approve` in LP moved outside after initialization of the LP and ercAddress needed to approve.

Reply:
You are right. `approve` in LP moved outside after initialization of the LP and ercAddress needed to approve.